### ISSUE TITLE
Bug: Using `from __future__ import annotations` breaks template support

### ISSUE THREAD

**Issue by Nadock** [starlite-api/starlite#1271]
**Describe the bug**
If you include `from __future__ import annotations` in the same file as a route handler that returns a `starlite.Template` the template will not be rendered. Instead, the `starlite.Template` object is converted to JSON and that is returned instead.

**To Reproduce**
Below is code that reproduces the error for me.

```python
# app.py
from __future__ import annotations

import pathlib

import starlite
from starlite.contrib import jinja


@starlite.get("/")
def index() -> starlite.Template:
    return starlite.Template(name="repro.html")


app = starlite.Starlite(
    route_handlers=[index],
    template_config=starlite.TemplateConfig(
        directory=pathlib.Path("./templates"),
        engine=jinja.JinjaTemplateEngine,
    ),
)
```

```html
<!-- templates/repro.html -->
<!DOCTYPE html>
<html lang="en-au">

<head>
    <meta content="text/html;charset=utf-8" http-equiv="Content-Type">
    <meta content="utf-8" http-equiv="encoding">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="color-scheme" content="dark light">
    <title>from __future__ import annotations</title>
</head>

<body>
    <h1>Hello, world!</h1>
</body>
```

**Additional context**
**With** the `from __future__ import annotations` line:
![Screenshot 2023-03-04 at 17 41 19](https://user-images.githubusercontent.com/1491530/222881622-8e8eebe7-52a4-4ba0-b59d-f3989f6ba9f0.png)

**Without** the `from __future__ import annotations` line:

![Screenshot 2023-03-04 at 17 46 14](https://user-images.githubusercontent.com/1491530/222881797-5e2d66a5-05b4-4a52-b0ea-0ba91c69aa2a.png)

---

I've only just started using **Starlite** so I'm not sure if there is something I missed in the documentation that says you shouldn't use `from __future__ import annotations`. But even if there is, maybe **Starlite** could raise an exception when you do this.

**Comment by Goldziher** [comment]
Hi, what version have you tested? And can you check our main branch?

**Comment by Nadock** [comment]
This was tested on `1.51.6` and I've also been able to reproduce this when installing 06c50d7 from `main`, all using Python `3.11.1`.

**Comment by peterschutt** [comment]
I actually came across the cause of this today, and it will be fixed for 2.0.

The handler gets a reference to its signature here:

https://github.com/starlite-api/starlite/blob/fbe51fb18bb685e579b66d0a42f1e0c3a4fae24f/starlite/handlers/http.py#L477

However, support for resolving types in `inspect.Signature.from_callable()` is only available in 3.10+.

We go on to use an `isinstance()` check on the return annotation, but it is still a string here:

https://github.com/starlite-api/starlite/blob/fbe51fb18bb685e579b66d0a42f1e0c3a4fae24f/starlite/handlers/http.py#L610

This will be fixed in 2.0 from this commit: https://github.com/starlite-api/starlite/pull/1160/commits/1ce9c5a84a0836139a2570317855b9a14d15d808 - and a few of the preceding commits on that branch.

We could do some `eval()` work to try to resolve the type of the return annotation in the 1.5x branch, but I'd suggest that you don't rely to heavily on `__future__.annotations` anywhere that there needs to be runtime introspection of types on the 1.x series.

**Comment by Nadock** [comment]
If there's a fix coming in `2.0` that's excellent 👏 

> We could do some eval() work to try to resolve the type of the return annotation in the 1.5x branch, but I'd suggest that you don't rely to heavily on __future__.annotations anywhere that there needs to be runtime introspection of types on the 1.x series.

Yeah that's fine honestly. This was my first foray into using Starlite so I the side effects of `__future__.annotations` wasn't clear. Maybe worth a note in the documentation that `__future__.annotations` can cause issues, but it doesn't sound like back porting a fix is that worth it.
